### PR TITLE
#162685660 User receives email upon successful registration

### PR DESCRIPTION
**Pull request by fahadmak**
#### What does this PR do?
This PR delivers a feature that enables a user to receive an email with an account activation upon successful registration.

#### Description of Task to be completed?
When a user suucessfuly signs up to the Authors Haven Application, he/she  should receive a verification email with a link to the application.

#### How should this be manually tested?

Before testing run these commands in cmd

`git clone https://github.com/andela/Ah-backend-guardians.git`

`cd Ah-backend-guardians`

`git checkout ft-email-confirmation-162685660`

`virtualenv env`

`. env/bin/activate`

`pip install -r requirements.txt`

1.  Create a postgres database and a `.env` file in your root directory and add the following configurations.
```
SECRET_KEY=andela_sjdndkjghdf78872

DB_NAME=your_database_name

DB_USER=database_user

DB_PASSWORD=database_password

DB_HOST=127.0.0.1

ALLOWED_HOSTS=.localhost, .herokuapp.com

EMAIL_PORT=587

EMAIL_HOST_USER = gauthorshaven@gmail.com

EMAIL_HOST_PASSWORD = guardiansauthors
```
2. Make sure an email account or if not create one. This account will receive the email activation link. 
3. Open Postman and type `http://localhost:8000/api/users/` in url text box.
4. In the body section of postman, type in the following json format. Note type in correct email address which will receive the link.
  {  
     "user":{
                    "email":"your_email_addresss",
                      "password": "your_password",
                      "username":"Greesn123d44ssfse"
                }
  }
5. Send a POST request to the url and then go to your email inbox to check wheter you have received the email.
6. click on the activation link to activate your account.

#### What are the relevant pivotal tracker stories?
#162685660

#### Screenshots (if appropriate)
screen shot 2019-01-15 at 12 26 41 pm

### CODE REVIEW

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -18,16 +28,60 @@ class RegistrationAPIView(APIView):
...
+        current_site = get_current_site(request)
+        response_data = self.send_email_notification(current_site, user)
+        response_data.update(serializer.data)
+        return Response(response_data, status=status.HTTP_201_CREATED)
+
+    def send_email_notification(self, current_site, user):
+        domain = f'http://{current_site.domain}'
+        token = default_token_generator.make_token(user)
+        uid = urlsafe_base64_encode(force_bytes(user.username)).decode('utf-8')
+        print(domain)
```
Comment (Opio-Emmanuel-Omona) [inline authors/apps/authentication/views.py]: please remove the print statements.

Comment (fahadmak) [reply authors/apps/authentication/views.py]: thanks for the feedback let me work onthat

File: authors/apps/authentication/views.py [hunk 2]
```diff
@@ -18,16 +28,58 @@ class RegistrationAPIView(APIView):
...
+
+    def send_email_notification(self, current_site, user):
+        domain = f'http://{current_site.domain}'
+        token = default_token_generator.make_token(user)
+        uid = urlsafe_base64_encode(force_bytes(user.username)).decode('utf-8')
+        route = reverse('authentication:confirm', args=[token, uid])
+        url = f'{domain}{route}'
+        subject = "Activate your Author's Haven Account."
+        message = f'You have successfully registered an AH account. \n Click \
+                    Link Below To Activate Your Account\n{url}'
```
Comment (Opio-Emmanuel-Omona) [inline authors/apps/authentication/views.py]: image

Fix this bug, there's too much space appearing in the message.

Comment (fahadmak) [reply authors/apps/authentication/views.py]: Thanks for pointing that out I have now fixed the issue

**Review by myrdstom — COMMENTED** [review]
Please test the status code in your tests.. otherwise LGTM

**Comment by BarnaTB** [comment]
@fahadmak please change the user object you're sending and adopt the one we agreed on which does not require the `user` key before passing the user's credentials. These changes were made earlier and that branch was merged so I advise that you rebase and get those latest changes to avoid having conflicts on `develop`

**Review by VivianDoreen — APPROVED** [review]
LGTM, I can receive a link to my email requesting me to activate the account, I receive a descriptive message when I try to log in with my account not activated, the tests are passing, the styling is upto standard

**Review by VivianDoreen — CHANGES_REQUESTED** [review]
Great work, 
The functionality is successful though it would be better if you reviewed areas you have not tested for such that coverage does not drop.

**Review by myrdstom — APPROVED** [review]
LGTM

**Review by VivianDoreen — APPROVED** [review]

**Review by Opio-Emmanuel-Omona — APPROVED** [review]
LGTM

**Review by tomuhenry — APPROVED** [review]
LGTM

**Review by archbishop12 — APPROVED** [review]
LGTM @fahadmak I am merging this.

**Pull request merged**